File: MH/DAY6/10845.js
const path = process.platform === "linux" ? "/dev/stdin" : "../input.txt";
let input = require("fs").readFileSync(path).toString().trim().split("\n");
const [n, ...commands] = input;

let queue = [];
let answer = [];

for (let i = 0; i < n; i++) {
  switch (commands[i]) {
    case "pop":
      answer.push(queue.shift() || -1);
      // if(queue.length === 0) answer.push(-1);
      // else answer.push(queue.shift());
      break;
    case "size":
      answer.push(queue.length);
      break;
    case "empty":
      answer.push(queue.length ? 0 : 1);
      // if(queue.length === 0) answer.push(1);
      // else answer.push(0);
      break;
    case "front":
      answer.push(queue[0] || -1);
      // if(queue.length === 0) answer.push(-1);
      // else answer.push(queue[0]);
      break;
    case "back":
      answer.push(queue[queue.length - 1] || -1);
      // if(queue.length === 0) answer.push(-1);
      // else answer.push(queue[queue.length-1]);
      break;
    default:
      queue.push(commands[i].split(" ")[1]);
      break;
  }
}

console.log(answer.join("\n"));

File: KSY/DAY8/10845.py
# import heapq
import sys
input = sys.stdin.readline

n = int(input())
queue = []
# heapq.heapify(queue)

for i in range(n):
    op = input().split()
    if op[0] == "push":
        queue.append(int(op[1]))
    elif op[0] == "pop":
        if queue:
            print(queue[0])
            del queue[0]
        else:
            print(-1)
    elif op[0] == "size":
        print(len(queue))
    elif op[0] == "empty":
        if queue:
            print(0)
        else:
            print(1)
    elif op[0] == "front":
        if queue:
            print(queue[0])
        else:
            print(-1)
    elif op[0] == "back":
        if queue:
            print(queue[-1])
        else:
            print(-1)
